### PR TITLE
add support for source schema and exclude/include pattern  and fix numeric type cast

### LINKED ISSUE #26: getting zlib error when running the update_tables
Zlib::GzipFile::Error: closed gzip stream

here is the stack trace
```/home/deploy/apps/vydia/shared/bundle/ruby/2.1.0/bundler/gems/postgres_to_redshift-1cf00b46e712/lib/postgres_to_redshift.rb:121:in `close'
/home/deploy/apps/vydia/shared/bundle/ruby/2.1.0/bundler/gems/postgres_to_redshift-1cf00b46e712/lib/postgres_to_redshift.rb:121:in `ensure in copy_table'
/home/deploy/apps/vydia/shared/bundle/ruby/2.1.0/bundler/gems/postgres_to_redshift-1cf00b46e712/lib/postgres_to_redshift.rb:122:in `copy_table'
/home/deploy/apps/vydia/shared/bundle/ruby/2.1.0/bundler/gems/postgres_to_redshift-1cf00b46e712/lib/postgres_to_redshift.rb:27:in `block in update_tables'
/home/deploy/apps/vydia/shared/bundle/ruby/2.1.0/bundler/gems/postgres_to_redshift-1cf00b46e712/lib/postgres_to_redshift.rb:24:in `each'
/home/deploy/apps/vydia/shared/bundle/ruby/2.1.0/bundler/gems/postgres_to_redshift-1cf00b46e712/lib/postgres_to_redshift.rb:24:in `update_tables'```

on ruby 2.1.9   directly using the master branch code  postgresql is on 9.4

### PR DESCRIPTION
**Pull request by dingn1**
Added the ability to specify source schema.Also added a way to only run for certain tables and only exclude certain tables. also fixed #26

### CODE REVIEW

**Comment by dingn1** [comment]
@toothrot remake the pr and added the spec need some help to get the spec to pass.

**Comment by dingn1** [comment]
@toothrot ?

**Comment by dingn1** [comment]
@toothrot ?

**Comment by toothrot** [comment]
Sorry for the delay, I will look at this shortly.

**Comment by dingn1** [comment]
No matter, thanks.

File: spec/features/small_table_spec.rb [hunk 1]
```diff
@@ -26,3 +26,33 @@
...
+        INSERT INTO "p2r_integration" ("title") VALUES ('Casablanca');
+          CREATE TABLE IF NOT EXISTS "exclude_table" ("id" SERIAL PRIMARY KEY, "value" text);
+          INSERT INTO "p2r_integration" ("value") VALUES ('Test');
+      EOS
+      )
+    end
+    after(:all) do
+      PostgresToRedshift::Test.test_connection.exec(%q[DROP TABLE IF EXISTS "p2r_integration";])
+      PostgresToRedshift::Test.test_target_connection.exec(%q[DROP TABLE IF EXISTS "p2r_integration";])
+
```
Comment (toothrot) [inline spec/features/small_table_spec.rb]: nit: remove empty line here

File: README.md [hunk 2]
```diff
@@ -24,13 +24,20 @@ Or install it yourself as:
...
 ```bash
 export POSTGRES_TO_REDSHIFT_SOURCE_URI='postgres://username:password@host:port/database-name'
 export POSTGRES_TO_REDSHIFT_TARGET_URI='postgres://username:password@host:port/database-name'
 export POSTGRES_TO_REDSHIFT_TARGET_SCHEMA='testing-data'
+export POSTGRES_TO_REDSHIFT_SOURCE_SCHEMA='testing-data'
 export S3_DATABASE_EXPORT_ID='yourid'
 export S3_DATABASE_EXPORT_KEY='yourkey'
 export S3_DATABASE_EXPORT_BUCKET='some-bucket-to-use'
+export DROP_TABLE_BEFORE_CREATE = true
+export POSTGRES_TO_REDSHIFT_EXCLUDE_TABLE_PATTERN = 'table-pattern-to-exclude1,table-pattern-to-exclude2'
```
Comment (toothrot) [inline README.md]: Maybe comment here that all tables are included by default?

File: README.md [hunk 3]
```diff
@@ -24,13 +24,20 @@ Or install it yourself as:
...
+
 ```bash
 export POSTGRES_TO_REDSHIFT_SOURCE_URI='postgres://username:password@host:port/database-name'
 export POSTGRES_TO_REDSHIFT_TARGET_URI='postgres://username:password@host:port/database-name'
 export POSTGRES_TO_REDSHIFT_TARGET_SCHEMA='testing-data'
+export POSTGRES_TO_REDSHIFT_SOURCE_SCHEMA='testing-data'
 export S3_DATABASE_EXPORT_ID='yourid'
 export S3_DATABASE_EXPORT_KEY='yourkey'
 export S3_DATABASE_EXPORT_BUCKET='some-bucket-to-use'
+export DROP_TABLE_BEFORE_CREATE = true
```
Comment (toothrot) [inline README.md]: Also worth mentioning that this is false by default

File: README.md [hunk 4]
```diff
@@ -24,13 +24,20 @@ Or install it yourself as:
...
 Set your source and target databases, as well as your s3 intermediary.
 
+*Note: Only set the POSTGRES_TO_REDSHIFT_EXCLUDE_TABLE_PATTERN if you want to exlude certain table(comma seperate for multiple value)*
+*Note: set DROP_TABLE_BEFORE_CREATE to true will drop the table on target before creation*
+
 ```bash
 export POSTGRES_TO_REDSHIFT_SOURCE_URI='postgres://username:password@host:port/database-name'
 export POSTGRES_TO_REDSHIFT_TARGET_URI='postgres://username:password@host:port/database-name'
 export POSTGRES_TO_REDSHIFT_TARGET_SCHEMA='testing-data'
+export POSTGRES_TO_REDSHIFT_SOURCE_SCHEMA='testing-data'
```
Comment (toothrot) [inline README.md:39]: Worth mentioning that this is 'public' by default

**Review by toothrot — COMMENTED** [review]
Looks good! Thanks!

**Comment by dingn1** [comment]
@toothrot just updated the readme added a bunch of notes

**Comment by flah00** [comment]
@dingn1 The env var docs you added include spaces between `=`, which will lead to bash errors. 

```bash
export DROP_TABLE_BEFORE_CREATE = true
```
Should be 

```bash
export DROP_TABLE_BEFORE_CREATE=true
```

**Comment by dingn1** [comment]
@toothrot bump

**Comment by dingn1** [comment]
@toothrot bump

**Pull request closed without merging**